Mage.Sets/src/mage/cards/p/PrimalMight.java

```diff
@@ -0,0 +1,78 @@
+package mage.cards.p;
+
+import mage.abilities.Ability;
+import mage.abilities.dynamicvalue.common.ManacostVariableValue;
+import mage.abilities.effects.ContinuousEffect;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.FightTargetsEffect;
+import mage.abilities.effects.common.continuous.BoostTargetEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.target.TargetPermanent;
+import mage.target.common.TargetControlledCreaturePermanent;
+import mage.target.targetpointer.FixedTarget;
+
+import java.util.UUID;
+
+/**
+ *
+ * @author jmharmon
+ */
+
+public final class PrimalMight extends CardImpl {
+
+    public PrimalMight(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{X}{G}");
+
+        // Target creature you control gets +X/+X until end of turn. Then it fights up to one target creature you don’t control.
+        this.getSpellAbility().addEffect(new PrimalMightEffect());
+        this.getSpellAbility().addTarget(new TargetControlledCreaturePermanent());
+        this.getSpellAbility().addTarget(new TargetPermanent(StaticFilters.FILTER_CREATURE_YOU_DONT_CONTROL));
```

Comment:
Must be up to one target (e.g. player can only boost creature or boost with fight). So it must be `0, 1` target -- I don't known how it will works with it. Please, try real card play to test it with zero and one targets (plus legal and non legal).